src/app/components/Page.tsx

```diff
@@ -32,20 +33,37 @@ export interface PageConfig {
     icon?: string;
 }
 
-interface Props {
+interface Props extends RouteComponentProps {
     config: PageConfig;
     children?: React.ReactNode;
 }
 
-const Page = ({ config, children }: Props) => {
+const Page = ({ config, location, children }: Props) => {
     const userPermissions = usePermissions();
     const { sections = [], permissions: pagePermissions = [], text } = config;
     const [activeSection, setActiveSection] = useState('');
+    const containerRef = useRef<HTMLDivElement>(null);
 
-    React.useEffect(() => {
+    useEffect(() => {
         document.title = `${text} - ProtonMail`;
     }, [text]);
 
+    useEffect(() => {
+        if (!location.hash) {
+            return;
+        }
+
+        // Need a delay to let the navigation ends
+        setTimeout(() => {
```

Comment:
You need to `clearTimeout` here with the callback for the `useEffect`.

Reply:
At least you are on the same page with @mmso 😄